app.js

```diff
@@ -0,0 +1,142 @@
+const unleash = require('unleash-server')
+const KeycloakStrategy = require('./lib/strategy.js')
+const passport = require('passport')
+const {
+  KC_HOST,
+  KC_REALM,
+  KC_CLIENT_ID,
+  KC_CLIENT_SECRET,
+  KC_ADMIN_ROLES,
+  KC_EDITOR_ROLES,
+  KC_VIEWER_ROLES,
+  ADMIN_ACCESS_TOKEN,
+  CLIENT_ACCESS_TOKEN
+} = require('./env')
+
+const enableKeycloakOauth = (app, config, services) => {
+  const { baseUriPath } = config.server
+  const { userService, accessService } = services
+
+  passport.use(
+    'keycloak',
+    new KeycloakStrategy(
+      {
+        host: KC_HOST,
+        realm: KC_REALM,
+        clientID: KC_CLIENT_ID,
+        clientSecret: KC_CLIENT_SECRET,
+        callbackURL: `${baseUriPath}/api/auth/callback`,
+        authorizationURL: `${KC_HOST}/realms/${KC_REALM}/protocol/openid-connect/auth`,
+        tokenURL: `${KC_HOST}/realms/${KC_REALM}/protocol/openid-connect/token`,
+        userInfoURL: `${KC_HOST}/realms/${KC_REALM}/protocol/openid-connect/userinfo`
+      },
+
+      async (accessToken, refreshToken, profile, done) => {
+        console.log(`username: ${profile.username} roles: ${profile.roles}`)
+
+        const getRole = (userRoles) => {
```

Comment:
Is it worth splitting this out and adding a test for getRole() as well?

Reply:
Yes, I can, just this test won't be the same as other tests using request, I haven't figured out how to test keycloak part properly.

Reply:
added in https://github.com/app-sre/unleash/pull/57/commits/ace23ed79dfaaf9eefdeacd5de28be302c4c58f5